src/r_util.cpp

```diff
@@ -478,6 +479,12 @@ namespace rhost {
             return R_NilValue;
         }
 
+        extern "C" SEXP get_package_lock_state(SEXP path) {
```

Comment:
This appears to be a generic function to check whether any given file (not just package) is locked - the name should reflect that.